articles/azure-functions/durable-functions-disaster-recovery-geo-distribution.md

```diff
@@ -0,0 +1,81 @@
+description: Learn about Disaster Recovery and Geo Distribution in Durable Functions.
+services: functions
+author: casard
+manager: cfowler
+editor: ''
+tags: ''
+keywords:
+ms.service: functions
+ms.devlang: multiple
+ms.topic: article
+ms.tgt_pltfrm: multiple
+ms.workload: na
+ms.date: 04/25/2018
+ms.author: azfuncdf
+---
+
+# Disaster Recovery and Geo-Distribution
+
+## Overview
+In Azure Durable Functions, all state is persisted in Azure Storage. A [task hub](durable-functions-task-hubs.md) is a logical container for Azure Storage resources that are used for orchestrations. Orchestrator and activity functions can only interact with each other when they belong to the same task hub.
+The scenarios described in this article propose deployment options to increase availability and minimize downtime during disaster recovery activities.
+It is important to notice that these scenarios are based on an Active-Passive configurations, since they are guided by the usage of Azure Storage.
+
+
+>[!NOTE]
+>All function apps that are redundantly deployed, must share the same function access keys in the case of being activated via HTTP. The Functions Runtime exposes a [management API](https://github.com/Azure/azure-functions-host/wiki/Key-management-API) that enables consumers to programmatically add, delete, and update function keys.
+>
+
+## Scenario 1 - Load Balanced Compute with Shared Storage
+In the event of problems of the compute service in Azure, the Function App could be affected. To minimize the possibility of such downtime, this scenario uses two instances of the Function App deployed to different regions. The underlying storage account and task hub are created in the main region, and it is shared by both instances.
+Traffic Manager is configured to detect problems in the main site and automatically redirect traffic to the healthy Function App in the failover region.
+Because Function App the failover region shares the same Azure Storage account and Task Hub, the state of the functions is not lost, and word can resume normally. Once the health is restored in the main region, Traffic Manager will start routing requests to that instance of the Function App automatically.
+
+
+![Diagram showing scenario 1.](media/durable-functions-disaster-recovery-geo-distribution/durable-functions-geo-scenario01.png)
+
+There are several benefits when using this deployment scenario:
+- In the case of a failure in the compute layer, work can resume in the failover region without state loss.
+- Traffic Manager takes care of the automatic failover to the healthy instance automatically.
+- Traffic Manager automatically re-establishes traffic to the main Function App after the outage has been corrected.
+
+However, consider the following when using this scenario:
+- If the Function App is deployed using a dedicated AppService plan, there is an increased cost by replicating the compute instance in the failover datacenter.
```

Comment:
there is an increased cost by replicating the compute instance in the failover datacenter.

-->

replicating the compute instance in the failover datacenter increases costs.

Reply:
check